src/ImageSharp/Formats/WebP/LosslessUtils.cs

```diff
@@ -0,0 +1,656 @@
+                newBlue += ColorTransformDelta((sbyte)m.RedToBlue, (sbyte)newRed);
+                newBlue &= 0xff;
+
+                pixelData[i] = (argb & 0xff00ff00u) | ((uint)newRed << 16) | (uint)newBlue;
+            }
+        }
+
+        /// <summary>
+        /// This will reverse the predictor transform.
+        /// The predictor transform can be used to reduce entropy by exploiting the fact that neighboring pixels are often correlated.
+        /// In the predictor transform, the current pixel value is predicted from the pixels already decoded (in scan-line order) and only the residual value (actual - predicted) is encoded.
+        /// The prediction mode determines the type of prediction to use. The image is divided into squares and all the pixels in a square use same prediction mode.
+        /// </summary>
+        /// <param name="transform">The transform data.</param>
+        /// <param name="pixelData">The pixel data to apply the inverse transform.</param>
+        /// <param name="output">The resulting pixel data with the reversed transformation data.</param>
+        public static void PredictorInverseTransform(Vp8LTransform transform, Span<uint> pixelData, Span<uint> output)
+        {
+            int processedPixels = 0;
+            int yStart = 0;
+            int width = transform.XSize;
+            Span<uint> transformData = transform.Data.GetSpan();
+
+            // First Row follows the L (mode=1) mode.
+            PredictorAdd0(pixelData, processedPixels, 1, output);
+            PredictorAdd1(pixelData, 1, width - 1, output);
+            processedPixels += width;
+            yStart++;
+
+            int y = yStart;
+            int yEnd = transform.YSize;
+            int tileWidth = 1 << transform.Bits;
+            int mask = tileWidth - 1;
+            int tilesPerRow = SubSampleSize(width, transform.Bits);
+            int predictorModeIdxBase = (y >> transform.Bits) * tilesPerRow;
+            while (y < yEnd)
+            {
+                int predictorModeIdx = predictorModeIdxBase;
+                int x = 1;
+
+                // First pixel follows the T (mode=2) mode.
+                PredictorAdd2(pixelData, processedPixels, 1, width, output);
+
+                while (x < width)
+                {
+                    uint predictorMode = (transformData[predictorModeIdx++] >> 8) & 0xf;
+                    int xEnd = (x & ~mask) + tileWidth;
+                    if (xEnd > width)
+                    {
+                        xEnd = width;
+                    }
+
+                    // There are 14 different prediction modes.
+                    // In each prediction mode, the current pixel value is predicted from one or more neighboring pixels whose values are already known.
+                    int startIdx = processedPixels + x;
+                    int numberOfPixels = xEnd - x;
+                    switch (predictorMode)
+                    {
```

Comment:
This is NOT TESTED and I'm quite unsure if it works, but:
all Predictor calls here have 4 identical parameters, so MAYBE the switch could be optimized as follows:

Those predictors that use the width as an additional parameter, could change their parameter order to use width as last parameter.
My very optimistic guess would be the following general code to be generated:

- push return address on stack
- push pixelData, startIndex, numberOfPixels and output on Stack,
- switch on predictorMode and depending on the value 
  - maybe push "width" to stack
  - jump to Predictor method entrypoint.

Maybe even adding width to ALL predictors and ignoring where not used could be beneficial.
This is an unchecked guess, so don't rely on it to work.